infrastructure/main.tf

```diff
@@ -103,6 +103,10 @@ module "recipe-database" {
 }
 
 module "recipe-database-v11" {
+  providers = {
+    azurerm             = azurerm
+    azurerm.private_dns = azurerm.private_dns
```

Comment:
you also need to define this provider.

it would be good to look at moving this to private endpoint as well as part of this change, you would need to pass the subnet_id variable and choose a subnet for it